### DISCUSSION TITLE
Swayamvar : A 2D Puzzle Game

### DISCUSSION THREAD

**Discussion by happyorsaad** [scaleracademy/open-source-projects#6] [Ideas]
### Summary 

It will be a 2D puzzle game where the protagonist, Arya, will be a prince archer whose aim would be to hit the eye of the fish at every level. The goal of the protagonist is to defeat levels of increasing difficulty and finally gain the right to be the husband of the prettiest girl in all of Amravati, Padmini. 

The idea is inspired by: https://gprosser.itch.io/sagittarius. But instead of just gravity, we can have other components. 
Think of it as a variation of mini-golf puzzles. Just set in a mythological setting.

### Procedural Level Generation

Each level will include 2 base entities, the protagonist and the target fish, both of whom will be stationary. The puzzle will then be generated procedurally depending on the level the player is currently at. 
The components can be sliding doors, a magical teleportation portal for the arrows, switches to open the arrow's path to the fish, etc
The player loses when he is not able to hit the fish using the allocated number of arrows.
 
### Multiplayer Face-Off

The game can have a multiplayer face-off feature where the players will be the target instead of the fish. Each player has to outsmart the other and fire the deadly arrow first.

Other features are just what we have in any other game, leaderboards, etc.

### Platform
The game can be built for any platform. If we use a game engine, we should be able to port this on multiple platforms.

**Comment by rajeev503** [comment]
Nice ideaa!!
